project-confero/confero
# Chores/awsdb

# Problem
While the current app is suitable for local development, we needed to make it available to be deployed on AWS Elastic-beanstalk running a single docker container.
 
# Solution
We run 3 commands created as a post app deployment hook ([H/T Edward Samuel](https://stackoverflow.com/questions/31978961/django-migrations-with-docker-on-aws-elastic-beanstalk?noredirect=1&lq=1)) which allows us to run the commands to the docker container after deployment - instead of before. These three commands:

1. Migrate the database
2. Create a superuser
3. Collect static file to be served

## Change summary

- Added ```.ebextensions/df-migrate.config``` to handle bash scripts to the docker container
- Added production environment variables for database connection and turned debugging off in the production environment for security reasons.
- Added ```fec/management/commands/createsu.py``` to create superusers programmatically and without input.

## Steps to Verify
1. Add environment admin user password to production environment
2. Add 
```
User.objects.create_superuser("<USER-NAME>", "<EMAIL>",
                                          os.environ['<ENVIRONMENT-VALUE>'])
```
to ```fec/management/commands/createsu.py```
3. ```eb deploy``` to environment 
4. Go to http://confero-env.cvqp3jgbbb.us-east-2.elasticbeanstalk.com/admin
5. Add Politicians and Campaigns
6. See Changes on homepage.

## Screenshot (Optional)
![image](https://user-images.githubusercontent.com/35119415/52766591-a39d5180-2ff5-11e9-9963-268783dad783.png)

# For Discussion
I couldn't figure out how to serve django static files, but that looks like it is [another ticket](https://stackoverflow.com/questions/7241688/django-admin-css-missing) [entirely](https://www.caktusgroup.com/blog/2014/11/10/Using-Amazon-S3-to-store-your-Django-sites-static-and-media-files/).
